src/goto-programs/goto_cfg.cpp

```diff
@@ -74,21 +77,33 @@ goto_cfg::goto_cfg(goto_functionst &goto_functions)
     }
 
     // Third pass - identify all the successors/predecessors
+    int counter = 0;
     for (auto &bb : bbs)
     {
       assert(bb->begin != bb->end);
       auto last = bb->end;
       last--;
+      bb->uuid = counter++;
 
       for (const auto &bb2 : bbs)
       {
+        const bool is_goto = last->is_goto() || last->is_backwards_goto();
+        const bool is_trivial_goto = is_goto && is_true(last->guard);
         if (bb2->begin == bb->end)
         {
-          bb->successors.insert(bb2);
-          bb2->predecessors.insert(bb);
+          if (is_trivial_goto)
```

Comment:
What do we define as a trivial goto?

Reply:
A goto with a trivial guard, so it only has one target in practice.